prow/cmd/generic-autobumper/bumper/bumper.go

```diff
@@ -891,3 +933,142 @@ func getTreeRef(stderr io.Writer, refname string) (string, error) {
+	var addedOptions []string
+	for _, v := range reviewers {
+		addedOptions = append(addedOptions, fmt.Sprintf("r=%s", v))
+	}
+	for _, v := range cc {
+		addedOptions = append(addedOptions, fmt.Sprintf("cc=%s", v))
+	}
+	if len(addedOptions) > 0 {
+		pushRef = fmt.Sprintf("%s%%%s", pushRef, strings.Join(addedOptions, ","))
+	}
+	return pushRef
+}
+
+func getDiff(prevCommit string) (string, error) {
+	var diffBuf bytes.Buffer
+	var errBuf bytes.Buffer
+	if err := Call(&diffBuf, &errBuf, gitCmd, "diff", prevCommit); err != nil {
+		return "", fmt.Errorf("error diffing previous bump: %v -- %s", err, errBuf.String())
+	}
+	return diffBuf.String(), nil
+}
+
+func GerritNoOpChange(changeID, hostRepo string) (bool, error) {
+	var garbageBuf bytes.Buffer
+	var outBuf bytes.Buffer
+	// Fetch current pending CRs
+	if err := Call(&garbageBuf, &garbageBuf, gitCmd, "fetch", "upstream", "+refs/changes/*:refs/remotes/upstream/changes/*"); err != nil {
+		return false, fmt.Errorf("unable to fetch upstream changes: %v", err)
+	}
+	// Get PR with same ChangeID for this bump
+	if err := Call(&outBuf, &garbageBuf, gitCmd, "log", "--all", fmt.Sprintf("--grep=Change-Id: %s", changeID), "-1", "--format=%H"); err != nil {
+		return false, fmt.Errorf("error getting previous bump: %v", err)
+	}
+	prevCommit := strings.TrimSpace(outBuf.String())
+	// No current CRs with cur ChangeID means this is not a noOp change
+	if prevCommit == "" {
+		return false, nil
+	}
+	diff, err := getDiff(prevCommit)
+	if err != nil {
+		return false, err
+	}
+	if diff == "" {
+		return true, nil
+	}
+	return false, nil
+
+}
+
+func createCR(msg, branch, changeID, hostRepo, cookieFile string, reviewers, cc []string, stdout, stderr io.Writer) error {
+	if err := Call(stdout, stderr, gitCmd, "config", "http.cookiefile", cookieFile); err != nil {
+		return fmt.Errorf("unable to load cookiefile: %v", err)
+	}
+	if err := Call(stdout, stderr, gitCmd, "remote", "add", "upstream", hostRepo); err != nil {
+		return fmt.Errorf("unable to add upstream remote: %v", err)
+	}
+	noOp, err := GerritNoOpChange(changeID, hostRepo)
+	if err != nil {
+		return fmt.Errorf("error diffing previous bump: %v", err)
+	}
+	if noOp {
+		logrus.Info("CR is a no-op change. Returning without pushing update")
+		return nil
+	}
+
+	pushRef := buildPushRef(branch, reviewers, cc)
+	if err := Call(stdout, stderr, gitCmd, "commit", "-a", "-v", "-m", msg); err != nil {
+		return fmt.Errorf("unable to commit: %v", err)
+	}
+	if err := Call(stdout, stderr, gitCmd, "push", "origin", pushRef); err != nil {
+		return fmt.Errorf("unable to push: %v", err)
+	}
+	return nil
+}
+
+func getLastBumpCommit(gerritAuthor, commitTag string) (string, error) {
+	var outBuf bytes.Buffer
+	var errBuf bytes.Buffer
+
+	if err := Call(&outBuf, &errBuf, gitCmd, "log", fmt.Sprintf("--author=%s", gerritAuthor), fmt.Sprintf("--grep=%s", commitTag), "-1", "--format='%H'"); err != nil {
+		return "", fmt.Errorf("Error finding old commit")
```

Comment:
I think this error is "error running git command", also you should use `errors.New` since there is no fmt

Reply:
Done